### PR TITLE
[ISSUE-222][BUG] RequestSlotResponse/RegisterShuffleResponse should handle null issue

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @FMX @waitinfuture

File: client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala [hunk 1]
```diff
@@ -271,6 +271,16 @@ class LifecycleManager(appId: String, val conf: RssConf) extends RpcEndpoint wit
         registerShuffleRequest.remove(shuffleId)
       }
       return
+    } else if (res.workerResource.isEmpty && numPartitions > 0) {
```
Comment (waitinfuture) [inline client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala:274]: Why the status is Success in this branch?

Comment (AngersZhuuuu) [reply client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala:274]: > Why the status is Success in this branch?

RequestSlots rerun success but with empty slots.

**Comment by waitinfuture** [comment]
LGTM, thanks!

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

File: client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala [hunk 2]
```diff
@@ -261,22 +261,34 @@ class LifecycleManager(appId: String, val conf: RssConf) extends RpcEndpoint wit
...
-          context.reply(RegisterShuffleResponse(StatusCode.SlotNotAvailable, null))
+    res.status match {
+      case StatusCode.Failed =>
+        logError(s"OfferSlots RPC request failed for $shuffleId failed!")
+        registerShuffleRequest.synchronized {
+          val set = registerShuffleRequest.get(shuffleId)
+          set.asScala.foreach { context =>
+            context.reply(RegisterShuffleResponse(StatusCode.Failed, List.empty.asJava))
+          }
+          registerShuffleRequest.remove(shuffleId)
```
Comment (waitinfuture) [inline client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala:272]: We should add a return in the end of the case branch?

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**